cpp/src/round/round.cu

```diff
@@ -191,13 +192,48 @@ struct half_even_negative {
 template <typename T>
 struct half_up_fixed_point {
   T n;
-  __device__ T operator()(T e) { return half_up_negative<T>{n}(e) / n; }
+  template <typename U = T, typename std::enable_if_t<cudf::is_floating_point<U>()>* = nullptr>
+  __device__ U operator()(U e)
+  {
+    assert(false);  // Should never get here. Just for compilation
+    return U{};
+  }
+
+  template <typename U = T, typename std::enable_if_t<cuda::std::is_integral<U>::value>* = nullptr>
+  __device__ U operator()(U e)
+  {
+    // Create a container with extra digit for adjustment
+    auto const container = e / n;
+    auto const down      = container / 10;
```

Comment:
Could you run some benchmarks on round before and after this change? Especially for DECIMAL128.  Divide is especially expensive for 128-bits and I want to be sure that these two new divides and the modulo are worth the cost compared to what we had before.  Expanding the range is great. I just don't want to be bitten by it in the common case.